recipe/cross_compile_support.sh

```diff
@@ -115,13 +116,19 @@ if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]]; then
             popd
             rm -rf ${EXTRACT_DIR}
         elif [[ "${CUDA_COMPILER_VERSION}" == "11.2" ]]; then
-	    echo "cross compiling with cuda == 11.2 and cdt != cos7/8 not supported yet"
-	    exit 1
+            echo "cross compiling with cuda == 11.2 and cdt != cos7/8 not supported yet"
+            exit 1
+        elif [[ "${CUDA_COMPILER_VERSION}" == "12.0" ]] && [[ "${CDT_NAME}" == "cos7" ]]; then
+            # Need libcuda.so.1 in test phase to test the executables
+            mv ./usr/local/cuda-${CUDA_COMPILER_VERSION}/compat/* ${QEMU_LD_PREFIX}/usr/lib/
```

Comment:
Not yet sure what we need to do here TBH. But the even the CUDA 12 setup doesn't contain `libcuda.so` anywhere, so I think this is still necessary. Downloading the artefacts shouldn't be necessary anymore, but we probably have to add dependencies on the cuda-compiler stack.

Reply:
```
mv: cannot stat ‘./usr/local/cuda-12.0/compat/*’: No such file or directory
```
We'll still have to install CUDA here somehow to get the driver I think

Reply:
So basically we just need a branch that doesn't `exit 1` for cuda 12?

Reply:
Yeah think so. It would be good if this works for any 12

Reply:
12 already works for native compilation, it's just that cross-compilation errors out for that currently

Reply:
Yeah we actually did some testing offline with cross-compilation and it seemed to work in the test cases we tried

The issues are this line needing to be changed and in some cases NVIDIA packages need GLIBC 2.28 at a minimum, which then become motivation for addressing issue ( https://github.com/conda-forge/conda-forge.github.io/issues/1941 ) in some fashion

That said, I think the Arrow case can skate by without resolving the GLIBC 2.28 (as its usage of CUDA packages is quite minimal). Though it is something we would want to address before migrating CUDA 12 arch packages generally

Reply:
Alright, I've set this to do nothing for CUDA 12 now.